File: spring/src/main/java/pl/kamil/poc/foo/FooDataEntity.java
package pl.kamil.poc.foo;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.*;

import javax.persistence.*;
import java.io.Serializable;

@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(name="foo_data")
public class FooDataEntity implements Serializable  {
    @Getter()
    @Setter()
    @Id()
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id = null;

    @Getter()
    @Setter()
    @Column()
    private String data = null;

    @JsonIgnore
    @Setter()
    @JoinColumn(name = "foo_id", nullable = false)
    @ManyToOne(fetch = FetchType.EAGER, optional = false)
    private FooEntity foo = null;

    @JsonIgnore
    public FooEntity getFoo() {
        return this.foo;
    }
}

File: nest/src/app/NestedEntityOptimisticLockingExample/data/FooData.entity.ts
import { Exclude } from 'class-transformer'
import { Column, Entity, JoinColumn, ManyToOne, PrimaryColumn } from 'typeorm'

import { FooEntity } from './Foo.entity'

@Entity({ name: 'foo_data' })
export class FooDataEntity {
  @PrimaryColumn({ generated: true })
  id!: number

  @Column()
  data: string

  @Exclude()
  @JoinColumn({ name: 'foo_id' })
  @ManyToOne(() => FooEntity, foo => foo.list)
  foo: FooEntity

  constructor(data: string, foo: FooEntity) {
    this.data = data
    this.foo = foo
  }
}
